pyhealth/datasets/base_dataset.py

```diff
@@ -386,17 +422,22 @@ def get_all_tokens(
             raise ValueError("Please set task first.")
         tokens = []
         for sample in self.samples:
+            if type(sample[key]) == list:
+                if len(sample[key]) == 0:
+                    continue
+                # a list of lists of values
+                elif type(sample[key][0]) == list:
+                    tokens.extend(sum(sample[key], []))
```

Comment:
I am not sure whether this `sum` operation works for concatenating two lists.
```
>>> sum([1,2,3], [])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: can only concatenate list (not "int") to list
```

Reply:
sum() flattens the nested list, e.g. [[1, 2], [3, 4]] -> [1, 2, 3, 4]

Reply:
Then, should we add one more bracket and use `sum([sample[key], []])`.

However, I also tried this
```
>>> sum([[1,2,3], []])
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: unsupported operand type(s) for +: 'int' and 'list'
```

Reply:
The correct usage is `sum([[1, 2], [3, 4]], start=[])`.

In the latest commit, [27ddaa7](https://github.com/sunlabuiuc/PyHealth/commit/27ddaa7254ba94fea47b9a27d0cfcefdbdc21be9), I wrapped this into a function `flatten_list()` for easier understanding.